apps/prairielearn/src/pages/instructorAssessments/instructorAssessments.ts

```diff
@@ -178,7 +204,33 @@ router.post(
   '/',
   asyncHandler(async (req, res) => {
     if (req.body.__action === 'add_assessment') {
-      const editor = new AssessmentAddEditor({ locals: res.locals });
+      if (!req.body.title) {
+        throw new HttpStatusError(400, 'title is required');
+      }
+      if (!req.body.aid) {
+        throw new HttpStatusError(400, 'aid is required');
+      }
+      if (!req.body.type) {
+        throw new HttpStatusError(400, 'type is required');
+      }
+      if (!req.body.set) {
+        throw new HttpStatusError(400, 'set is required');
+      }
+      if (!/^[-A-Za-z0-9_/]+$/.test(req.body.aid)) {
+        throw new HttpStatusError(
+          400,
+          `Invalid aid (was not only letters, numbers, dashes, slashes, and underscores, with no spaces): ${req.body.aid}`,
+        );
+      }
```

Comment:
Tiniest of nits: move this to immediately follow the previous `if (!req.body.qid) ...` check?

```suggestion
      if (!req.body.aid) {
        throw new HttpStatusError(400, 'aid is required');
      }
      if (!/^[-A-Za-z0-9_/]+$/.test(req.body.aid)) {
        throw new HttpStatusError(
          400,
          `Invalid aid (was not only letters, numbers, dashes, slashes, and underscores, with no spaces): ${req.body.aid}`,
        );
      }
      if (!req.body.type) {
        throw new HttpStatusError(400, 'type is required');
      }
      if (!req.body.set) {
        throw new HttpStatusError(400, 'set is required');
      }
```

Reply:
Might want to consider #9908 in this evaluation.